rdr_service/tools/tool_libs/consents.py

```diff
@@ -80,6 +86,60 @@ def _call_server_for_retro_validation(self, participant_ids):
         if response.status_code != 200:
             exit(1)  # Exiting program to prevent further calls with a bad key
 
+    def files_for_update(self):
+        logger.info('Generating file...')
+        storage_provider = GoogleCloudStorageProvider()
+        with open('output.csv', 'w') as output_file, self.get_session() as session:
+            # Updatable query to get the files to verify and modify
+            files: Collection[ConsentFile] = session.query(ConsentFile).join(
+                Participant, Participant.participantId == ConsentFile.participant_id
+            ).filter(
+                ConsentFile.type == ConsentType.EHR,
+                ConsentFile.sync_status == ConsentSyncStatus.NEEDS_CORRECTING
+            ).order_by(ConsentFile.participant_id, ConsentFile.file_upload_time).limit(500).all()
+
+            fields_to_output = [
```

Comment:
All of these fields come from the ConsentFile row?  Is the join to the Participant table intended to filter out "problem" ConsentFile data?

Reply:
Yes, all the fields come from the ConsentFile. No, the join there is an artifact of an earlier version of the query. This query is going to be a point of change and will likely be modified for each run of the script to get different sets of files that need to be looked into further.